go-controller/pkg/factory/handler.go

```diff
@@ -165,11 +221,16 @@ func (i *informer) addHandler(internalInformerIndex int, id uint64, priority int
 		handlerAlive,
 		priority,
 		internalInformerIndex,
+		atomic.Bool{},
+		make([]func(), 0, len(existingItems)+200),
```

Comment:
200 should be a constant, also it would be nice to explain that optimization only works for the first 200 events, so if you get more than that same full lock will happen

Reply:
yeah I just randomly chose 200, I can make it a constant, and maybe expose a metric and we can come up with a better value here.